tests/unit/s2n_self_talk_session_resumption_test.c

```diff
@@ -549,9 +553,29 @@ int main(int argc, char **argv)
         EXPECT_EQUAL(server_conn->actual_protocol_version, S2N_TLS13);
         EXPECT_EQUAL(client_conn->actual_protocol_version, S2N_TLS13);
 
-        EXPECT_SUCCESS(s2n_connection_free(server_conn));
-        EXPECT_SUCCESS(s2n_connection_free(client_conn));
-        EXPECT_SUCCESS(s2n_io_pair_close(&io_pair));
+        /* since TLS 1.3 was negotiated, and the client hasn't called recv yet
+         * there should be no session ticket available. 512 is an arbitrary
+         * number, nothing should be copied because this operation should fail
+         */
+        EXPECT_EQUAL(s2n_connection_get_session_length(client_conn), 0);
+        /* In the event that get_session actually writes data, we want an error,
+         * not a segfault.
+         */
+        s2n_realloc(&ticket, 512);
+        EXPECT_EQUAL(s2n_connection_get_session(client_conn, ticket.data, 512), 0);
```

Comment:
Minimally for safety this should really be:
```suggestion
        s2n_realloc(&ticket, 512);
        EXPECT_EQUAL(s2n_connection_get_session(client_conn, ticket.data, ticket.size), 0);
```
But also, if 512 is just an arbitrary number, why realloc with it instead of using the existing value of ticket.size?